app/src/main/java/com/example/drawingactivity/CustomColorViewModel.kt

```diff
@@ -84,6 +76,43 @@ class CustomColorViewModel : ViewModel() {
         }
     }
 
+    /**
+     * Removes a layer from the list of layers
+     */
+    fun deleteLayer(LayerName: String) {
+        layerList.remove(LayerName)
+    }
+
+    var newName: String? = null
+    var oldName: String? = null
+    var isEditBottomSheetOpen = false
+    var isEditDialogOpen = false
+    var onEditComplete: (() -> Unit)? = null
+
+    /**
+     * Edits the name of the layer
+     */
+    fun editLayer(layerName: String) {
```

Comment:
why is the viewModel holding a function like this, isn't the ViewModel just supposed to hold the state.